velox/exec/prefixsort/PrefixSortEncoder.h

```diff
@@ -53,35 +51,47 @@ class PrefixSortEncoder {
   }
 
   /// Encodes String types.
-  /// The string prefix is formatted as 'null byte + string content + padding
-  /// zeros'. If `!ascending_`, the bits for both the content and padding zeros
-  /// need to be inverted.
+  /// The first byte of the encoded result is null byte. The value is 0 if
+  /// (nulls first and value is null) or (nulls last and value is not null).
+  /// Otherwise, the value is 1.
   FOLLY_ALWAYS_INLINE void encode(
       std::optional<StringView> value,
       char* dest,
       int32_t encodeSize) const {
     if (value.has_value()) {
       dest[0] = nullsFirst_ ? 1 : 0;
-      std::string storage;
-      auto data = HashStringAllocator::contiguousString(value.value(), storage);
-      if (data.size() < encodeSize - 1) {
-        std::memcpy(dest + 1, data.data(), data.size());
-        std::memset(dest + 1 + data.size(), 0, encodeSize - 1 - data.size());
-      } else {
-        std::memcpy(dest + 1, data.data(), encodeSize - 1);
-      }
-
-      if (!ascending_) {
-        for (auto i = 1; i < encodeSize; ++i) {
-          dest[i] = ~dest[i];
-        }
-      }
+      encodeNoNulls(value.value(), dest, encodeSize);
```

Comment:
Please align with other `encodeNulls`, encode the value to dest and encode the value as `encodeSize`, so use encodeNoNulls(value.value(), dest + 1, encodeSize - 1);

Reply:
Updated, thanks